src/coreComponents/physicsSolvers/fluidFlow/SinglePhaseHybridFVMKernels.hpp

```diff
@@ -27,375 +27,346 @@
 #include "finiteVolume/mimeticInnerProducts/QuasiTPFAInnerProduct.hpp"
 #include "finiteVolume/mimeticInnerProducts/QuasiRTInnerProduct.hpp"
 #include "finiteVolume/mimeticInnerProducts/SimpleInnerProduct.hpp"
-#include "linearAlgebra/interfaces/InterfaceTypes.hpp"
 #include "mesh/MeshLevel.hpp"
 #include "physicsSolvers/fluidFlow/FlowSolverBaseExtrinsicData.hpp"
 #include "physicsSolvers/fluidFlow/HybridFVMHelperKernels.hpp"
+#include "physicsSolvers/fluidFlow/SinglePhaseBaseExtrinsicData.hpp"
+#include "physicsSolvers/fluidFlow/SinglePhaseBaseKernels.hpp"
+#include "physicsSolvers/fluidFlow/StencilAccessors.hpp"
 
 namespace geosx
 {
 namespace singlePhaseHybridFVMKernels
 {
 
-/******************************** AssemblerKernelHelper ********************************/
+/******************************** Kernel switches ********************************/
+
+namespace internal
+{
+
+template< typename T, typename LAMBDA >
+void kernelLaunchSelectorFaceSwitch( T value, LAMBDA && lambda )
+{
+  static_assert( std::is_integral< T >::value, "KernelLaunchSelectorFaceSwitch: type should be integral" );
```

Comment:
To be honest, I don't know how to compile with 
```
template< unsigned int NUM_FACES, typename LAMBDA >
```
instead of
```
template< typename T, typename LAMBDA >
```
I am really a dummy with templates, so I could not figure out which modifications were needed to the dispatch to make it compile. But I agree that your suggestions would be better.

Reply:
We can't of course make `NUM_FACES` a template parameter since the whole point of this function is runtime dispatch on `value`. The function was originally written with a parameter `T` so that we didn't have to commit to a specific integral type - we used to mix `integer`, `localIndex`, etc. in various ways, and whatever type the user specified would go back into the lambda as a compile-time constant. But we could also just commit to `integer` everywhere and remove `T`.

Reply:
> We can't of course make NUM_FACES a template parameter since the whole point of this function is runtime dispatch on value. 

Of course, sorry. 😵‍💫